src/mono/mono/eventpipe/ep-rt-mono.h

```diff
@@ -1063,9 +1063,10 @@ uint8_t *
 ep_rt_valloc0 (size_t buffer_size)
 {
 	uint8_t *buffer = (uint8_t *)mono_valloc (NULL, buffer_size, MONO_MMAP_READ | MONO_MMAP_WRITE, MONO_MEM_ACCOUNT_PROFILER);
-
-	if (buffer)
-		memset (buffer, 0, buffer_size);
+#ifdef EP_CHECKED_BUILD
+	for (size_t i = 0; i < buffer_size; i++)
+		EP_ASSERT (buffer [i] == 0);
```

Comment:
This is probably slower than just calling `memset`.

If you need to assert, it would be better to:
1. allocate a static global buffer of some large size (maybe `mono_pagesize`)
2. do a `memcmp` in pagesize chunks: `allzero = allzero && !memcmp (buffer+offset, zerobuf, 4096)`
3. Assert at the end that `allzero` is true

Also I'm not sure if this assert belongs in eventpipe or in mono/utils - in the various `mono_valloc` implementations.

Reply:
This is just for debug/checked builds, no need to make it more complicated Put it there since EventPipe did the memset, could be moved down as well, but not sure its worth doing another CI round for that now.

Reply:
ok